src/components/select/select.spec.ts

```diff
@@ -1,86 +1,86 @@
 import { expect, fixture } from '@open-wc/testing';
 import { html } from 'lit/static-html.js';
 
-import '../option';
-import './select';
 import { isSafari } from '../core/dom';
+import { waitForLitRender } from '../core/testing';
+import { testA11yTreeSnapshot } from '../core/testing/a11y-tree-snapshot';
+
+import type { SbbSelectElement } from './select';
+import './select';
+import '../option';
 
 describe('sbb-select', () => {
-  it('renders', async () => {
-    const root = await fixture(html`
-      <sbb-select>
-        <sbb-option value="1">Option 1</sbb-option>
-        <sbb-option value="2">Option 2</sbb-option>
-        <sbb-option value="3">Option 3</sbb-option>
-      </sbb-select>
-    `);
-    const listboxAttr = 'id="sbb-select-1" role="listbox"';
-
-    expect(root.shadowRoot!.host).to.have.attribute('dir', 'ltr');
-    expect(root.shadowRoot!.host).to.have.attribute('data-state', 'closed');
-    expect(root).shadowDom.to.be.equal(`
-      <div class="sbb-select__trigger" aria-hidden="true">
-        <span class="sbb-select__trigger--placeholder"></span>
-      </div>
-      <div class="sbb-select__gap-fix"></div>
-      <div class="sbb-select__container">
-        <div class="sbb-select__gap-fix">
-          <div class="sbb-gap-fix-wrapper">
-            <div class="sbb-gap-fix-corner" id="left"></div>
-          </div>
-          <div class="sbb-gap-fix-wrapper">
-            <div class="sbb-gap-fix-corner" id="right"></div>
-          </div>
-        </div>
-        <div class="sbb-select__panel">
-          <div class="sbb-select__wrapper">
-            <div class="sbb-select__options" ${!isSafari() ? listboxAttr : ''}>
-              <slot></slot>
-            </div>
-          </div>
-        </div>
-      </div>
-    `);
+  describe('renders', () => {
+    let elem: SbbSelectElement;
+
+    beforeEach(async () => {
+      elem = await fixture(html`
+        <sbb-select>
+          <sbb-option value="1">Option 1</sbb-option>
+          <sbb-option value="2">Option 2</sbb-option>
+          <sbb-option value="3">Option 3</sbb-option>
+        </sbb-select>
+      `);
+      await waitForLitRender(elem);
+    });
+
+    (!isSafari() ? describe : describe.skip)('Chrome-Firefox', async () => {
+      it('Dom', async () => {
+        await expect(elem).dom.to.be.equalSnapshot();
+      });
+
+      it('ShadowDom', async () => {
+        await expect(elem).shadowDom.to.be.equalSnapshot();
+      });
+    });
+
+    (isSafari() ? describe : describe.skip)('Safari', async () => {
+      it('Dom', async () => {
+        await expect(elem).dom.to.be.equalSnapshot();
+      });
+
+      it('ShadowDom', async () => {
+        await expect(elem).shadowDom.to.be.equalSnapshot();
+      });
+    });
```

Comment:
I saw this a couple of time - what's the issue with Safari?

Reply:
Some components have a different template or behavior depending on the browser. In those cases, we cannot use snapshots unless we separate them by browser. And essentially this is what I did here